apps/modernization-ui/src/apps/page-builder/services/conditionAPI.ts

```diff
@@ -1,12 +1,38 @@
 /* eslint-disable camelcase */
-import { Condition, ConditionControllerService, CreateConditionRequest } from '../generated';
+import { Condition, ConditionControllerService, CreateConditionRequest, ReadConditionRequest } from '../generated';
 
-export const fetchConditions = (token: string): Promise<Array<Condition>> => {
-    return ConditionControllerService.findAllConditionsUsingGet({
+export const fetchConditions = async (token: string): Promise<Array<Condition>> => {
+    const response = await ConditionControllerService.findAllConditionsUsingGet({
         authorization: token
-    }).then((response) => {
-        return response;
     });
+    return response;
+};
+
+export const findConditions = async (token: string, page: number, size: number, sort: string) => {
+    const response = await ConditionControllerService.findConditionsUsingGet({
+        authorization: token,
+        page,
+        size,
+        sort
+    });
+    return response;
+};
+
+export const searchConditions = async (
+    token: string,
+    page: number,
+    size: number,
+    sort: string,
+    search: ReadConditionRequest
+) => {
+    const response = await ConditionControllerService.searchConditionsUsingPost({
+        authorization: token,
+        search,
+        page,
+        size,
+        sort
+    });
+    return response;
 };
```

Comment:
The auto generated method is returning `Page_Condition_ | any`, can we have this API method call return `Page_Condition_`. This way we can use the type provided. I am not sure why the generated code is using any. I think its an issue with the Java `Page` interface.